src/custom_minimizer.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Feb  5 13:00:53 2016
+
+@author: SamWitte
+
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.optimize import minimize
+
+
+"""
+The general idea here is this:
+Can I create a minimization procedure that is more efficient than that standard basinhopping (and finds the actual minimium).
+To do this, I plan on picking my value of vmin, minimizing wrt to the value of eta (this can be done very quickly,
+this is the advantage of this method), then minimizing at a fixed
+eta across different vmin, then reminimize wrt to eta, etc.
+
+Currently this only functions for a single extended likelihood (CDMS-Si) and experiments capable of utilizing a poisson likelihood
+"""
+
+
+def Custom_SelfConsistent_Minimization(class_name, x0, mx, fp, fn, delta, vminStar=None, logetaStar=None,
+                                       index=None, vmin_err=8.0, logeta_err=0.01):
+
+        if vminStar is not None:
+            vmin_list_reduced = x0[: len(x0)/2]
+            vmin_list = np.sort(np.append(vmin_list_reduced, vminStar))
+
+            index_hold = np.argwhere(vmin_list == vminStar)[0, 0]
+            logeta_list_reduced = x0[len(x0)/2:]
+            logeta_list = np.insert(logeta_list_reduced, index_hold, logetaStar)
+        else:
+            vmin_list = x0[: len(x0)/2]
+            logeta_list = x0[len(x0)/2:]
+            logeta_list_reduced = logeta_list
+            vmin_list_reduced = vmin_list
+            index_hold = None
+
+        def constr_func_logeta(x):
+
```

Comment:
Don't put empty lines right after function definition, makes it harder to read the code.